### ISSUE TITLE
CA2000 with Marshal.GetIUnknownForObject should not warn

### ISSUE THREAD

**Issue by drewnoakes** [dotnet/roslyn-analyzers#4099]
### Analyzer

**Diagnostic ID**: [CA2000](https://docs.microsoft.com/visualstudio/code-quality/CA2000): `Dispose objects before losing scope`

### Analyzer source

**NuGet Package**: [Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

**Version**: 3.3.0

### Describe the bug

```c#
public IntPtr M()
{
    var disposable = new MyDisposableClass();
    //               ^^^ gives CA2000 Dispose objects before losing scope

    return Marshal.GetIUnknownForObject(disposable);
}
```

I'm assuming the lifetime of the object doesn't take the conversion to ppunk into account.

**Comment by Evangelink** [comment]
I am not really familiar with this API but I guess we should not only prevent the analyzer for reporting here but also prevent the disposing of the object, don't we?
